Add guide on connecting to Databases (#1527)
vaadin/docs: articles/integrations/databases/mysql.adoc

---
title: MySQL
description: Use MySQL in your Vaadin application.

> order: 10

Reviewer: ```suggestion
order: 30
```

layout: page
---

= Connecting to MySQL

> == Adding MySQL to an existing Vaadin project

Reviewer: ```suggestion
```

As with MariaDB, I think this heading is superfluous since there is no other option.

For an existing Vaadin project, you can connect to MySQL by following the steps below:

. Add the `mysql-connector-java` dependency to the [filename]#pom.xml# file.
+
.`pom.xml`
[source, xml]
----
    <dependency>
      <groupId>mysql</groupId>
      <artifactId>mysql-connector-java</artifactId>
      <scope>runtime</scope>
    </dependency>
----

> . Add configuration properties for MySQL to the [filename]#application.properties# file. 

Reviewer: ```suggestion
. Add MySQL configuration properties to the [filename]#src/main/resources/application.properties# file.
```

> You will have to replace `HOST`, `PORT`, `DB_NAME`, `USERNAME`, `PASSWORD` with corresponding parameters from your MySQL instance.

Reviewer: I have the same thought as the one I mentioned with MariaDB: let's add the defaults and mention that people should review them as a caveat.

Author: The only default is root without a password, which is bad to use as an example.
